debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -143,39 +234,71 @@ public static long discontiguousHeaderSize()
+				return VoidPointer.cast(J9IndexableObjectWithDataAddressContiguousCompressedPointer.cast(objPointer).dataAddr());
 			}
-			return VoidPointer.cast((J9IndexableObjectContiguousFullPointer.cast(objPointer)).dataAddr());
+			return VoidPointer.cast(J9IndexableObjectWithDataAddressContiguousFullPointer.cast(objPointer).dataAddr());
 		}
-		return VoidPointer.cast((J9IndexableObjectContiguousPointer.cast(objPointer)).dataAddr());
+		return VoidPointer.cast(J9IndexableObjectWithDataAddressContiguousPointer.cast(objPointer).dataAddr());
 	}
 
 	/**
-	 * @param objPointer the discontiguous array object who's dataAddr field we are accessing
-	 * @throws CorruptDataException If there's a problem accessing the indexable object dataAddr field
-	 * @throws NoSuchFieldException If the indexable object dataAddr field does not exist on the build that generated the core file
+	 * Get the dataAddr field for the discontiguous array object.
+	 *
+	 * @param objPointer the discontiguous array object whose dataAddr field we are accessing
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @throws NoSuchFieldException if the indexable object dataAddr field does not exist on the build that generated the core file
 	 */
 	public static VoidPointer getDataAddrForDiscontiguous(J9IndexableObjectPointer objPointer) throws CorruptDataException, NoSuchFieldException
 	{
 		if (mixedReferenceMode) {
 			if (compressObjectReferences) {
-				return VoidPointer.cast((J9IndexableObjectDiscontiguousCompressedPointer.cast(objPointer)).dataAddr());
+				return VoidPointer.cast(J9IndexableObjectWithDataAddressDiscontiguousCompressedPointer.cast(objPointer).dataAddr());
 			}
-			return VoidPointer.cast((J9IndexableObjectDiscontiguousFullPointer.cast(objPointer)).dataAddr());
+			return VoidPointer.cast(J9IndexableObjectWithDataAddressDiscontiguousFullPointer.cast(objPointer).dataAddr());
 		}
-		return VoidPointer.cast((J9IndexableObjectDiscontiguousPointer.cast(objPointer)).dataAddr());
+		return VoidPointer.cast(J9IndexableObjectWithDataAddressDiscontiguousPointer.cast(objPointer).dataAddr());
+	}
+
+	/**
+	 * Get the dataAddr field for the indexable object.
+	 *
+	 * @param objPointer the array object whose dataAddr field we are accessing
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @throws NoSuchFieldException if the indexable object dataAddr field does not exist on the build that generated the core file
+	 */
+	public static VoidPointer getDataAddrForIndexable(J9IndexableObjectPointer objPointer) throws CorruptDataException, NoSuchFieldException
+	{
+		return isInlineContiguousArraylet(objPointer)
+			? getDataAddrForContiguous(objPointer)
+			: getDataAddrForDiscontiguous(objPointer);
 	}
 
 	/**
-	 * @param objPointer array object who's elements we are outputting to dst
+	 * @param objPointer array object whose elements we are outputting to dst
```

Comment:
I don't know what this means and it doesn't seem to match the name or behavior of this method: What is `dst`, for example? Perhaps it should say "array object whose elements we are accessing"?

Reply:
I don't know either. This was modified to be consistent with your "who's/whose" typo comment in a method I wrote, as this is where I originally copied my code from which led to the misspelling in the method I added.